src/main/groovy/org/pih/warehouse/monitoring/SentryGrailsTracingFilter.groovy

```diff
@@ -0,0 +1,26 @@
+package org.pih.warehouse.monitoring
+
+import io.sentry.HubAdapter
+import io.sentry.IHub
+import io.sentry.spring.tracing.SentryTracingFilter
+
+/**
+ * Creates HTTP request transactions for Sentry performance tracing.
+ *
+ * Note that this bean is registered in grails-app/conf/spring/resource.groovy
+ *
+ * Because Grails and Spring MVC have different approaches for mapping URIs to controllers, we need to generate
+ * the transaction name ourselves for traces to display properly in Sentry.
+ *
+ * https://forum.sentry.io/t/sentry-java-spring-boot-and-grails-performance-tracing-not-working/15765/3
+ */
+class SentryGrailsTracingFilter extends SentryTracingFilter {
+
+    SentryGrailsTracingFilter() {
+        this(HubAdapter.getInstance())
+    }
+
+    SentryGrailsTracingFilter(final IHub hub) {
+        super(hub, new GrailsHttpTransactionNameProvider())
```

Comment:
I assume here we could use SpringMvcTransactionNameProvider to get the default behavior. Does the default behavior use the URL mapping?

Reply:
yes but it doesn't work with grails (nothing will ever get logged to sentry) so we'd need to provide our own implementation regardless. See that forum post linked in the class docstring if you're curious